Mathlib/LinearAlgebra/QuadraticForm/Basis.lean

```diff
@@ -91,4 +91,33 @@ noncomputable def toBilinHom (bm : Basis ι R M) : QuadraticMap R M N →ₗ[S]
   map_add' := add_toBilin bm
   map_smul' := smul_toBilin S bm
 
+open LinearMap.BilinMap (toQuadraticMap_apply)
+
+lemma toBilin_symm_eq_polarBilin (Q : QuadraticMap R M N) (bm : Basis ι R M) :
+    Q.toBilin bm + (Q.toBilin bm).flip = polarBilin Q := by
+  rw [← LinearMap.BilinMap.polarBilin_toQuadraticMap, toQuadraticMap_toBilin]
+
+theorem toBilin_toQuadraticMap (B : BilinMap R M N) (bm : Basis ι R M) (x y : M) :
+    let s := (bm.repr x).support ∪ (bm.repr y).support
+    B.toQuadraticMap.toBilin bm x y =
+      ∑ i ∈ s,
+        bm.repr x i • bm.repr y i • B (bm i) (bm i) +
+      ∑ p ∈ s.offDiag with p.1 < p.2,
+        bm.repr x p.1 • bm.repr y p.2 • (B + B.flip) (bm p.1) (bm p.2) := by
```

Comment:
This lemma feels very contrived to me

Reply:
Okay, I'll close this PR. Sorry to have wasted your time.